src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs

```diff
@@ -45,7 +51,18 @@ public static ProjectInfo CreateProjectInfo(this ProjectFileInfo projectFileInfo
                 language: LanguageNames.CSharp,
                 filePath: projectFileInfo.FilePath,
                 outputFilePath: projectFileInfo.TargetPath,
-                compilationOptions: projectFileInfo.CreateCompilationOptions());
+                compilationOptions: projectFileInfo.CreateCompilationOptions(),
+                analyzerReferences: analyzerReferences);
+        }
+
+        private static IEnumerable<AnalyzerReference> ResolveAnalyzerReferencesForProject(ProjectFileInfo projectFileInfo, IAnalyzerAssemblyLoader analyzerAssemblyLoader)
+        {
+            foreach(var analyzerAssemblyPath in projectFileInfo.Analyzers.Distinct())
+            {
+                analyzerAssemblyLoader.AddDependencyLocation(analyzerAssemblyPath);
+            }
+
+            return projectFileInfo.Analyzers.Select(analyzerCandicatePath => new AnalyzerFileReference(analyzerCandicatePath, analyzerAssemblyLoader));
```

Comment:
How is this used differently than the logic in CachingCodeFixProvider to load analyzers?

Reply:
Code fixes and analyzers can come from same assemblies, but analyzers are given to Project object and CodeFixes are cached elswhere since project doesn't have any information about it's CodeFixes available at model itself. This has same assembly ref part but after that they are different, extension method creates AnalyzerFileReference that are handled by roslyn internally and codeFixes per project are fetched from this cache. I am not 100% sure is there some other way to do this but it seems roslyn doesn't expose means for it at this point. If i remember correctly there are kind of similar mechanism inside of roslyn source code, however this memory goes almost year back so who knows 🙂 I think this is something which can be refactored if better means show up.